app/queries/payments_requiring_action.rb

```diff
@@ -6,8 +6,8 @@ def initialize(user)
   end
 
   def query
-    Spree::Payment.joins(order: [:user]).where.not(cvv_response_message: nil).
-      where("spree_users.id = ?", user.id)
+    Spree::Payment.joins(order: [:user]).where("spree_users.id = ?", user.id).
```

Comment:
Do you need to join `users`? `orders.user_id` should be enough, right?

It may be useful to tell ActiveRecord that Payment has_one user via order. Then you could probably just write `where(user_id: user.id)` here.

Reply:
yeah! good catch about the unnecessary join

Reply:
I tried adding the association and then changing this to `where(user_id: user.id)`, but then it expects there to be a column on Payment called `user_id`... so I just removed the unnecessary join.